homeassistant/components/elkm1/__init__.py

```diff
@@ -187,7 +187,7 @@ async def _async_discovery(*_: Any) -> None:
             hass, await async_discover_devices(hass, DISCOVER_SCAN_TIMEOUT)
         )
 
-    asyncio.create_task(_async_discovery())
+    hass.async_create_task(_async_discovery())
```

Comment:
```suggestion
    hass.async_create_background_task(_async_discovery())
```

This one will always run for 10s waiting for responses.  Its only needed to update the IP in the rare event it changes so it cna run in the background